src/Datadog.Trace.ClrProfiler.Native/stats.h

```diff
@@ -1,22 +1,29 @@
 #ifndef DD_CLR_PROFILER_STATS_H_
 #define DD_CLR_PROFILER_STATS_H_
 
-#include "util.h"
 #include <chrono>
 
+#include "util.h"
+
 namespace trace {
 
 class SWStat {
-  std::chrono::nanoseconds *_value;
+  std::chrono::nanoseconds* _value;
+  std::mutex* _mutex;
   std::chrono::steady_clock::time_point _startTime;
 
  public:
-  SWStat(std::chrono::nanoseconds *value) {
-      _value = value;
-      _startTime = std::chrono::steady_clock::now();
+  SWStat(std::chrono::nanoseconds* value, std::mutex* mutex) {
+    _value = value;
+    _mutex = mutex;
+    _startTime = std::chrono::steady_clock::now();
   }
   ~SWStat() {
-    *_value += std::chrono::steady_clock::now() - _startTime;
+    auto increment = std::chrono::steady_clock::now() - _startTime;
+    {
+      std::lock_guard<std::mutex> guard(*_mutex);
+      *_value += increment;
+    }
```

Comment:
Would it be possible to replace all these mutexes with `std::atomic<T>::fetch_add`? That's in C++11 - can't remember what version we target?

Reply:
Sadly no, at least not directly. I tried but we can't use `std::chrono::time_point` as a template instance, due compilation error, I guess one way to do it is to convert `std::chrono::time_point` to `long long` every time.

Reply:
trying this approach in db0520d